src/classes/Contacts.cls

```diff
@@ -0,0 +1,216 @@
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @description Domain class for Contacts in NPSP.
+*/
+public inherited sharing class Contacts extends fflib_SObjects {
+
+    private Map<Id, Contact> oldMap;
+    @TestVisible
+    private HouseholdSettings settings {
+        get {
+            if (settings == null) {
+                settings = new HouseholdSettings();
+            }
+            return settings;
+        }
+        set;
+    }
+
+    public Contacts(List<Contact> records) {
+        this(records, null);
+    }
+
+    public Contacts(List<Contact> records, List<Contact> oldList) {
+        super(records, Contact.SObjectType);
+        this.oldMap = oldList != null ?
+                new Map<Id, Contact>(oldList) : new Map<Id, Contact>();
+    }
+
+    public void onBeforeInsert() {
+        synchronizeNamingExclusions();
+    }
+
+    public void onBeforeUpdate() {
+        applyExclusionsStringChangesToCheckboxes();
+    }
+
+    private void applyExclusionsStringChangesToCheckboxes() {
+        for (Contact con : (List<Contact>) getRecords()) {
+            HouseholdNamingExclusionsString namingExclusionsString =
+                    new HouseholdNamingExclusionsString(namingExclusionsFor(con));
+            HouseholdNamingExclusionsString oldNamingExclusionsString =
+                    new HouseholdNamingExclusionsString(namingExclusionsFor(oldVersionOf(con)));
+            if (!namingExclusionsString.equals(oldNamingExclusionsString)) {
+                HouseholdNamingExclusions exclusionsAsCheckboxes = new HouseholdNamingExclusions(con);
+                exclusionsAsCheckboxes.updateFrom(namingExclusionsString);
+                setExclusionsCheckboxes(con, exclusionsAsCheckboxes);
+            }
+        }
+    }
+
+    public void onAfterInsert() {
+        if (orgUsesLegacyHouseholds()) {
+            ContactsInLegacyHouseholds contactsInLegacyHouseholds =
+                    new ContactsInLegacyHouseholds(getRecords());
+            contactsInLegacyHouseholds.onAfterInsert();
+        }
+    }
+
+    public void onAfterUpdate() {
+        new ContactsInLegacyHouseholds(getRecords(), oldMap)
+                .onAfterUpdate();
+    }
+
+    public void onAfterDelete() {
+        fireRollupsForMergedContacts();
+        List<Contact> contactsInHouseholds = contactsInHouseholds();
+        if (!contactsInHouseholds.isEmpty()) {
+            new ContactsInLegacyHouseholds(contactsInHouseholds)
+                    .onAfterDelete();
+        }
+    }
+
+    private Boolean orgUsesLegacyHouseholds() {
+        return settings.isAllProcessor() || settings.isAllIndividualsProcessor();
+    }
+
+    private Id householdIdFor(Contact con) {
+        return con.npo02__Household__c;
+    }
+
+    public void synchronizeNamingExclusions() {
```

Comment:
This method probably needs a block comment to explain what it's doing and why. The same true for `mergeExclusions` and some other related ones.